document-store/src/main/java/org/hypertrace/core/documentstore/model/subdoc/SubDocument.java

```diff
@@ -14,6 +15,7 @@
 @AllArgsConstructor(access = PRIVATE)
 public class SubDocument {
   private static final Set<String> IMPLICIT_PATHS = Set.of(CREATED_TIME, LAST_UPDATED_TIME);
+  private static final Pattern ALLOWED_CHARACTERS = Pattern.compile("^[a-zA-Z_]+(.[a-zA-Z_]+)*$");
```

Comment:
This is just a pre-check we are doing to ensure that the sub document path is always valid.
Well, it can contain numeric values. But, currently, there is no use-case for numeric values. So, decided to start with the strictest (so that we don't run into backward compatibility issues later). Whenever there is a use-case, we could relax the conditions here.